pallets/lottery/src/lib.rs

```diff
@@ -0,0 +1,351 @@
+//! # Lottery Monetization Module
+//!
+//! The Monetization module by lottery method
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency},
+};
+use frame_system::{self as system, ensure_signed};
+use log;
+use sp_runtime::{
+	sp_std::convert::TryInto,
+	traits::{One, Zero},
+	RuntimeDebug,
+};
+use sp_std::{cmp::Ordering, collections::btree_map::BTreeMap, ops::Add, prelude::*};
+
+use pallet_posts::Post;
+use pallet_utils::PostId;
+
+// TODO: move all tests to df-integration-tests
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+/// The pallet's configuration trait.
+pub trait Config: system::Config + pallet_posts::Config {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Config>::Event>;
+
+	type Currency: Currency<Self::AccountId>;
+}
+type PostVotesNumber = u64;
+
+pub type VoteKey<T: Config> = (T::BlockNumber, T::AccountId, PostId, PostVotesNumber);
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub struct LotteryResults<T: Config> {
+	winner_voter: Option<T::AccountId>,
+	winner_posts: Vec<(PostId, PostVotesNumber)>,
+	spent: bool,
+}
+
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub enum LotteryStatus<T: Config> {
+	Done(LotteryResults<T>),
+	InProgress,
+}
+
+impl<T: Config> Default for LotteryStatus<T> {
+	fn default() -> Self {
+		LotteryStatus::InProgress
+	}
+}
+impl<T: Config> LotteryStatus<T> {
+	pub fn is_done(&self) -> bool {
+		return match self {
+			LotteryStatus::Done(_) => true,
+			LotteryStatus::InProgress => false,
+		};
+	}
+}
+pub const VOTER_SHARE: u64 = 20;
+pub const NUMBER_OF_WINNING_POSTS: u64 = 3;
+
+// This pallet's storage items.
+decl_storage! {
+	trait Store for Module<T: Config> as LotteryModule {
+	 /// An id for the next donation.
+
+	pub VoterPricePercentage get(fn voter_price_share): u64 = VOTER_SHARE;
+	pub NumberOfWinningPosts get(fn number_of_winning_posts): u64 = NUMBER_OF_WINNING_POSTS;
```

Comment:
Make these as constants.
No need to make them storage values as they're used as read-only.